external/esp_idf_port/esp32/esp32_queue_api.c

```diff
@@ -0,0 +1,434 @@
+ *	item - The received message.
+ *	block_time_tick - the time (in Ticks) to wait until a timeout is decleared
+ * Return:
+ *	Returns pdPASS if success, pdFAIL if failure.
+ *
+ ****************************************************************************/
+int32_t IRAM_ATTR queue_recv_wrapper(void *queue, void *item, uint32_t block_time_tick)
+{
+	queue_info_t *queue_info = NULL;
+	size_t msglen = 0;
+	int prio = 0;
+	int32_t ret;
+	struct timespec abstime;
+	clock_t msecs;
+	clock_t secs;
+	clock_t nsecs;
+
+	if (!queue || !item) {
+		return pdFAIL;
+	}
+    queue_info = (queue_info_t *) queue;
+	if (queue_info->mqd_fd_recv != (mqd_t) ERROR) {
+		msglen = queue_info->mq_item_size;
+		if (block_time_tick == 0xFFFFFFFF) {
+			ret = mq_receive(queue_info->mqd_fd_recv, (char *)item, msglen, &prio);
+			if (ret == ERROR) {
+				return pdFAIL;
+			}
+		} else {
+			(void)clock_gettime(CLOCK_REALTIME, &abstime);
+			msecs = TICK2MSEC(block_time_tick);
+			secs = msecs / MSEC_PER_SEC;
+			nsecs = (msecs - (secs * MSEC_PER_SEC)) * NSEC_PER_MSEC;
+			abstime.tv_sec += secs;
+			abstime.tv_nsec += nsecs;
+			ret = mq_timedreceive(queue_info->mqd_fd_recv, (char *)item, msglen, &prio, &abstime);
+			if (ret == ERROR) {
+				return pdFAIL;
+			}
+		}
+	} else {
+		return pdFAIL;
+	}
+	return pdPASS;
+}
+
+/****************************************************************************
+ * Name: queue_recv_from_isr_wrapper
+ *
+ * Description:
+ *	This function is the wrapper of mq_timedreceive to receives the oldest of the highest priority
+ *messages from the message queue in an interrupt handle.
+ *
+ * Inputs:
+ *	queue - The queue to receive messages.
+ *	item - The received message.
+ *	hptw - useless in TizenRT, but needed in the declaration of esp32 wifi os adapter.
+ * Return:
+ *	Returns pdPASS if success, pdFAIL if failure.
+ *
+ ****************************************************************************/
+int32_t IRAM_ATTR queue_recv_from_isr_wrapper(void *queue, void *item, int32_t *const hptw)
+{
+	queue_info_t *queue_info = NULL;
+	size_t msglen = 0;
+	int prio = 0;
+	int32_t ret;
+
+	if (!queue || !item) {
+		return pdFAIL;
+	}
+	queue_info = (queue_info_t *) queue;
+	if (queue_info->mqd_fd_recv != (mqd_t) ERROR) {
+		msglen = queue_info->mq_item_size;
+		ret = mq_tryreceive_isr(queue_info->mqd_fd_recv, (char *)item, msglen, &prio);
+		if (ret == ERROR) {
+			return pdFAIL;
+		}
+		return pdPASS;
+	}
+	return pdFAIL;
+}
+
+/****************************************************************************
+ * Name: queue_msg_waiting_wrapper
+ *
+ * Description:
+ *	This function gets the message count of message queue.
+ *
+ * Inputs:
+ *	queue - The queue to send messages.
+ * Return:
+ *	Returns the count of messages in the queue if success,  returns pdFAIL if failure.
+ *
+ ****************************************************************************/
+uint32_t IRAM_ATTR queue_msg_waiting_wrapper(void *queue)
```

Comment:
please check if uint32_t or int32_t here

Reply:
it return number of msg in queue or failed value, 
as this is interface mainly used for wifi driver, 
it define fail value to 0,
so this function will be not return negtive value.